test/unit_test/passes/onnx/test_conversion.py

```diff
@@ -178,3 +180,79 @@ def mock_onnx_export_func(*args, **kwargs):
     p = create_pass_from_dict(OnnxConversion, {}, disable_search=True)
     _ = p.run(input_model, str(output_folder))
     assert "past_key_values" in dummy_inputs  # pylint: disable=unsupported-membership-test
+
```

Comment:
not directly related to this PR but should we also parameterize dynamo export in `test_onnx_conversion_pass` at line 24?

Reply:
Done. But awkwardly, dynamo exporter fails to export hf_model because the nested structure of past_key_values. It also involves mismatches input_names, which kind of says it's impossible to turn Optimum config to dynamo exporter required format in automation. I will bring this to the team, and see how should we proceed this. But the test passes, because we have fallback system from dynamo_exporter to torchscript exporter. As long as the test passes torchscript exporter, it will pass dynamo_exporter.

Reply:
Do you think it's failing because of what we discussed in the other comment https://github.com/microsoft/Olive/pull/1511#discussion_r1879064104?

That part is still unaddressed. For the optimum produced io_config + automatic cache, it is still missing the dynamic shapes for the optimum part. 
If you also created dynamic shapes there instead of just skipping as is now, it might work? We already have the dynamic axes, so the dynamic shapes creation there should be straightforward too.

Reply:
It should be just CI torch version. https://github.com/microsoft/Olive/blob/dc6700880d4df013fcfe49f2b6da59c9955ca14f/olive/passes/onnx/conversion.py#L206

Fallback should cover the case.

Reply:
What about locally, does the dynamo exporter test with hf model work without the fallback? Because we definitely want that case to work with these dynamic shapes changes.

Reply:
It works without fallback with this PR: https://github.com/pytorch/pytorch/pull/143158. 

However, there are some cases it won't work, as `dynamic_axes` creates no constraints, while `dynamic_shapes` examines the models while exporting them. We will fix them one by one when we see the user cases.